tst/testinstall/bitfields.tst

```diff
@@ -60,7 +60,7 @@ Error, Field getter: <data> must be a small integer (not a ffe)
 gap> bf.setters[1](1, (1,2));
 Error, Field Setter: <val> must be a small integer (not a permutation (small))
 gap> bf.setters[1]([],1);
-Error, Field Setter: <data> must be a small integer (not a list (plain,empty))
+Error, Field Setter: <data> must be a small integer (not a empty plain list)
```

Comment:
And here is the unfortunate side effect of your nice changes. "not a empty..." (or "not a immutable..." or whatever), when the correct English would be "not an empty..." (or "not an immutable..."). As an English speaker, this is something that mildly upsets me, but I can live with this. Do we try to make sure that indefinite articles (a/an) are correct in GAP? I'd guess not. I can imagine that it would be a pain to make this always work properly, so I'm not going to suggest that we should.

Reply:
It probably wouldn't be that hard to fix :) We could have an investigate.

Reply:
Actually, making this right was (for me) one of the driving motivations (yes, seriously :-) for introducing those `Require*` helpers (changed `not a integer` to `not an integer` and then later `not the integer %d`). So, I'd really want to improve this, at least when using the `Require` macros. One trick option here would be to modify `RequireArgumentEx`. Like this:
```patch
diff --git a/src/error.c b/src/error.c
index ff156757a..f7b071f35 100644
--- a/src/error.c
+++ b/src/error.c
@@ -590,8 +590,19 @@ Obj RequireArgumentEx(const char * funcname,
     else if (op == Fail)
         strlcat(msgbuf, " (not the value 'fail')", sizeof(msgbuf));
     else {
-        strlcat(msgbuf, " (not a %s)", sizeof(msgbuf));
-        arg1 = (Int)TNAM_OBJ(op);
+        const char * tnam = TNAM_OBJ(op);
+        // check for starting vowel
+        // TODO: move this into a separate helper function?
+        // TODO: improve this, and e.g. use "a" before "un"; "an hour"; ...
+        //  see also <https://github.com/rossmeissl/indefinite_article/blob/master/lib/indefinite_article.rb>
+        //  alternatively, could enhance the TNAM records to (optionally) allow indicating whether 'an' or 'a'
+        //  should be used....
+        if (strchr("aeiou", tnam[0]))
+            strlcat(msgbuf, " (not an", sizeof(msgbuf));
+        else
+            strlcat(msgbuf, " (not a", sizeof(msgbuf));
+        strlcat(msgbuf, tnam, sizeof(msgbuf));
+        strlcat(msgbuf, ")", sizeof(msgbuf));
     }
 
     ErrorMayQuit(msgbuf, arg1, arg2);
```